lib/usbd/backend/usbd_dwc_otg.c

```diff
@@ -173,11 +173,11 @@ static void disable_all_non_ep0(usbd_device *dev)
 
 		REBASE(DWC_OTG_DOEPxINT, i) = 0xFFFF;
 		REBASE(DWC_OTG_DOEPxTSIZ, i) = 0;
-		REBASE(DWC_OTG_DOEPxCTL, i) = DWC_OTG_DOEPCTL_SNAK;
+		REBASE(DWC_OTG_DOEPxCTL, i) = DWC_OTG_DOEPCTL_SNAK | DWC_OTG_DOEPCTL_EPDIS;
```

Comment:
That is why a NAK has been placed on the endpoint (to mark as not an active endpoint).
host is not expected to IO from the endpoint.
Also, who is handling the endpoint disable interrupt callback?